camel/CSVExample/src/main/java/ibmi/example/CSVExample.java

```diff
@@ -0,0 +1,178 @@
+         * below
+         */
+        final String filelocation = "tmp/";
+
+
+        /**
+         * Standard for a Camel deployment. Start by getting a CamelContext object.
+         */
+        CamelContext context = new DefaultCamelContext();
+
+        /**
+         * This sets up the connection for local IBM i Here is where we will store
+         * connection information for the application
+         */
+        AS400JDBCDataSource localDS = new AS400JDBCDataSource("localhost", "*CURRENT", "*CURRENT");
+
+        /**
+         * These values below exist if you are not journaling your IBM i table that you are querying
+         */
+        //localDS.setTransactionIsolation("none");
+        
+        /**
+         * This binds the localDS we set above to the route jt400
+         */
+        context.getRegistry().bind("jt400", localDS);
+
+        /**
+         * Setting our Marshall options for CSV
+         */
+		
+
+        /**
+         * What example do we want to accomplish? Set the variable above to choose the example 
+         */
+        context.addRoutes(new RouteBuilder() {
+            @Override
+            public void configure() {
+                from("timer://ibmiexamplecheckinterval?period=5000")
+                        /**
+                         * This is our example query. Here we are pulling results from the system.
+                         * You set the body of our program (which is a mutable object used to contain the results throughout).
+                         * Here I have set it to a constant, but you can use the simple() function to set some dynamic values
+                         * if needed in your own example. Once you have set the body, you send to JT400. The results are then 
+                         * stored in our body, replacing the query we set it to before.  
+                         */
+                          .setBody(constant("select * from QSYS2.NETSTAT_INFO"))
+                          .to("jdbc:jt400")
+                        /**
+                         * Now we have queried some results! We want to send it to a CSV for output. 
+                         */                
+                        .to("direct:" + whatexample);
+            }
+        });
+
+        /**
+         * This will do a standard output of values to CSV
+         */
+        context.addRoutes(new RouteBuilder() {
+            @Override
+            public void configure() {
+                // The from should match our to in the calling route. This connects our two routes
+                from("direct:standardcsvexample")
+                // Marshall is how we are going to divide up each result from the row and what to do
+                .marshal()
+                // This creates our CSV from the results that were marshalled
+                .csv()
+                // This specific header we are setting is the name of our file
+                .setHeader("CamelFileName", constant("CSVExample.csv"))
```

Comment:
Instead of a hardcoded filename, should we use something like `messageTimestamp` with the simple expression language? Or a rotating integer? As written, the same CSV will be overwritten constantly (I think)

Alternatively, we could set `fileExist=Append` on the file endpoint, and change the SQL to include `CURRENT_TIMESTAMP` in one of the columns (this might require declaring all the columns on the SQL)

Reply:
I can make an example that does a bit more like this